combinator_logic_experiments/y_comb_examples.metta

```diff
@@ -0,0 +1,57 @@
+(= (I $x) $x)
+(= ((K $x) $y) $x)
+(= (K $x $y) $x)
+(= (S $x $y $z) ($x $z ($y $z)))
+
+(= (B $x $y $z) ($x ($y $z)))
+(= (C $x $y $z) ($x $y $z))
+(= (W $x $y) ($x $y $y))
+
+(= U (S I I))
+
+(= Y (S (K (S I I)) (S (S (K S) K) (K (S I I)))))
+
+(= Y (B U (C B U)))
+
+
+; want to check equivalence between SKI and BCKW systems
```

Comment:
Oleg, I believe it is needed to work with application of combinators more accurately. In code above `(S $x $y $z)` is in fact `(((S $x) $y) $z)`. Using first notation one cannot point MeTTa interpreter how to interpret `(S (K S) K x y z)` it is just not matched. In papers this short notation is used for a sake of clarity. But more formally one need to translate it to the chain of combinators application. 

For example rewriting  SKI combinators like this:
```
(= (I $x) $x)
(= ((K $x) $y) $x)
(= (((S $x) $y) $z) (($x $z) ($y $z)))
```
one can make the following asserts work:
```
; want to check equivalence between SKI and BCKW systems
!(assertEqual
  (((((S (K S)) K) x) y) z)
   (x (y z))) ; (B x y z)

!(assertEqual
    (((((S ((S (K ((S (K S)) K))) S)) (K K)) x) y) z)
    ((x z) y)) ; (C x y z)

!(assertEqual
    ((((S S) (S K)) x) y)
    ((x y) y)) ; (W x y)
```
I commented out BCKV combinators applications because they also should obey this rule.
I needed to rewrite all the applications as bi-tuples to allow MeTTa interpreter apply application correctly.
If we want to keep clean notation I believe we need to write combinators interpreter, some function `(calc-combinators (S (K S) K x y z))` which obeys rules and returns `(x (y z))` it doesn't seems a complex task on a first glance.